support/sier-codec/src/schema.rs

```diff
@@ -243,4 +285,129 @@ mod tests {
             assert_eq!(struct_a.id(), struct_b.id());
         }
     }
+
+    #[cfg(test)]
+    mod parsing {
+        use super::*;
+
+        #[cfg(test)]
+        mod string {
+            use super::*;
+
+            #[test]
+            fn empty_string() {
+                let field = FieldDef {
+                    name: "foo".to_string(),
+                    type_: Type::String,
+                };
+
+                assert_eq!(
+                    field.parse(&[0]).unwrap(),
+                    (&[][..], Value::String("".to_string()))
+                );
+            }
+
+            #[test]
+            fn simple_string() {
+                let field = FieldDef {
+                    name: "foo".to_string(),
+                    type_: Type::String,
+                };
+
+                assert_eq!(
+                    field.parse(&[3, 65, 66, 67]).unwrap(),
+                    (&[][..], Value::String("ABC".to_string()))
+                );
+            }
+
+            #[test]
+            fn non_utf8() {
+                let field = FieldDef {
+                    name: "foo".to_string(),
+                    type_: Type::String,
+                };
+
+                assert!(field.parse(&[4, 0, 159, 146, 150]).is_err());
+            }
+
+            #[test]
+            fn keeps_extra_bytes() {
+                let field = FieldDef {
+                    name: "foo".to_string(),
+                    type_: Type::String,
+                };
+
+                assert_eq!(
+                    field.parse(&[2, 65, 66, 67, 68]).unwrap(),
+                    (&[67, 68][..], Value::String("AB".to_string()))
+                );
+            }
+
+            #[test]
+            fn no_size_byte() {
+                let field = FieldDef {
+                    name: "foo".to_string(),
+                    type_: Type::String,
+                };
+
+                assert!(field.parse(&[]).is_err());
+            }
+
+            #[test]
+            fn fewer_bytes_than_size_says() {
+                let field = FieldDef {
+                    name: "foo".to_string(),
+                    type_: Type::String,
+                };
+
+                assert!(field.parse(&[3, 65, 66]).is_err());
```

Comment:
Same as the previous comment, but for the `TooFewBytes` error.

Reply:
Done.